src/network/getters/getTokenAndBalanceForSponsorship.ts

```diff
@@ -15,7 +15,11 @@ export const getTokenAndBalanceForSponsorship = async (
     const tokenSymbolFromConfig = getCoreConfig().sponsorshipPaymentToken as string
     const tokenInformation = await getTokenInformation(
         defaultChainConfig.contracts[tokenSymbolFromConfig],
+        defaultChainConfig.id,
     )
+
+    // TODO it fails if user in not on the default chain (local or Polygon on prod) - needs refactor
+
```

Comment:
Won't `networkPreflight` solve it?

Reply:
Probably - but the issue is that we are calling it while displaying the list of sponsorships - I made a ticket about it so we can eventually discuss how to approach it on our Monday call

Reply:
`getTokenInformation` uses a chain-specific public node. Chain the user is on should not matter.